csi/src/freezefs.rs

```diff
@@ -0,0 +1,73 @@
+//! The files system freeze support using linux utility fsfreeze
+use crate::{dev::Device, mount};
+use snafu::Snafu;
+use tokio::process::Command;
+use uuid::Uuid;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub(crate)")]
+pub enum FreezeFsError {
+    #[snafu(display("Cannot find volume: {}", volid))]
+    VolumeNotFound { volid: String },
+    #[snafu(display("Invalid volume ID: {} {}", volid, error))]
+    InvalidVolumeId { volid: String, error: String },
+    #[snafu(display("fsfreeze failed for volume: {}, {}", volid, fserror))]
+    FsfreezeFailed { volid: String, fserror: String },
+    #[snafu(display("Internal failure for volume: {}, {}", volid, error))]
+    InternalFailure { volid: String, error: String },
+    #[snafu(display("IO error {}", error))]
+    IOError { error: String },
+}
+
+impl From<std::io::Error> for FreezeFsError {
+    fn from(error: std::io::Error) -> FreezeFsError {
+        FreezeFsError::IOError {
+            error: format!("{}", error),
+        }
+    }
+}
```

Comment:
If you create an appropriate FreezeFsError variant for std::Io:Error, there is no real need for this.

Reply:
Done. https://github.com/openebs/Mayastor/pull/462/commits/7c714899d911a2f659d5a7861968ba08b1aa354b